BackEndFlask/models/utility.py

```diff
@@ -14,3 +22,49 @@ def get_courses_by_user_courses_by_user_id(user_id):
     except SQLAlchemyError as e:
         error = str(e.__dict__['orig'])
         return error
+
+def get_users_by_course_id_and_role_id(course_id, role_id):
+    try:
+        users = []
+        users_and_role_ids = db.session.query(
+            User,
+            UserCourse.role_id
+        ).join(
+            UserCourse,
+            User.user_id == UserCourse.user_id
+        ).filter_by(
+            course_id=course_id
+        )
+        if role_id is not None:
```

Comment:
Why does a function for querying by role_id need a case for not havinga  role_id? Besides having role_id == None checks for a null role_id in SQL.

Reply:
The reason I make sure that role_id is not None is because we only want to filter the retrieved users if a role_id is passed. We do not want to filter the users by role_id of None for we actually just want to retrieve all users.